tests/net/lib/lwm2m/functional/pytest/test_lwm2m.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import time
+import logging
+import pytest  # noqa # pylint: disable=unused-import
+from leshan import Leshan
+import re
+import os
+import binascii
+from coapthon.client.helperclient import HelperClient
+import random
+import string
+
+from twister_harness.device.device_abstract import DeviceAbstract
+
+logger = logging.getLogger(__name__)
+
+class Shell(object):
+    def __init__(self, dut) -> None:
+        self._dut = dut
+
+    def find(self, regex):
+        pattern = re.compile(regex)
+        matched = False
+        timeout = 5  # [sec]
+        time_end = time.time() + timeout
+        while time.time() < time_end:
+            line = next(self._dut.iter_stdout)
+            if line is None or len(line) == 0:
+                continue
+            print(line)
+            m = pattern.match(line)
+            if m is not None:
+                return m
+
+        return None
+
+    def cmd(self, cmd):
+        self._dut.write(b'log halt\n')
+        self._dut.write(bytes(cmd + '\n', 'utf-8'))
+        found = False
+        answer = None
+        timeout = 5  # [sec]
+        time_end = time.time() + timeout
+        while time.time() < time_end:
+            line = next(self._dut.iter_stdout)
```

Comment:
Now you can find some problems when calling `iter_stdout` multiple times (it is a Generator). This PR #58491 should fix it